meta-mender-core/classes/mender-state-scripts.bbclass

```diff
@@ -0,0 +1,85 @@
+inherit deploy
+
+MENDER_STATE_SCRIPTS_DIR = "${B}/mender-state-scripts"
+
+# Default is to look in these two directories for scripts.
+MENDER_STATE_SCRIPTS ?= "${S}/mender-state-scripts ${MENDER_STATE_SCRIPTS_DIR}"
+
+do_create_default_script_dirs() {
+    # Doing this automatically has two advantages: The user can install directly
+    # into the directories with no preparation. And, we can complain if any of
+    # the targets in MENDER_STATE_SCRIPTS doesn't exist, because a directory
+    # that exists, but is empty, is different from not existing at all.
+
+    mkdir -p ${S}/mender-state-scripts ${B}/mender-state-scripts
```

Comment:
I think not, actually. The intention is that the variable can point to a nonexistent directory, and this should be an error. So we can't create it or the error would be silenced. This `mkdir` is specifically about our two defaults only.